### ISSUE TITLE
Compressed refs mode on AArch64 macOS

### ISSUE THREAD

**Issue by knn-k** [eclipse-openj9/openj9#13767]
You cannot build and run OpenJ9 in the compressed refs mode on AArch64 macOS.

OpenJ9 for x86-64 macOS is built with `-pagezero_size 0x1000` for mapping some memory in the first 32-bit address space.
On the other hand, AArch64 macOS does not allow executables with small page zero.  Running such binary causes a "Malformed Mach-o file" error.

**Comment by DanHeidinga** [comment]
If we had to use a `base + offset` or `base + shifted(offset)` model, would that affect our ability to stack allocate objects?  Scalarization would be unaffected but cases were we create an honest-to-goodness on stack object might be.

My current assumption is stack allocation **would** be affected even though java stacks would continue to be allocated from the suballocator as the user of the object would need to know which `base` to apply for a given object - the `heap`'s `base` or the suballocator's `base`.

If we allocated the suballocator and the heap with the same base, this wouldn't be an issue in practice but there may be other knock on effects.

**Comment by 0xdaryl** [comment]
@DanHeidinga : yes, there are situations where a SA or a HA object could be stored to a field and it would be impossible to distinguish what kind of object it is without further runtime checks.  If accessing such objects required different bases then that would be less than ideal.  In fact, I would go as far to say that it should be a design requirement that the bases are the same otherwise performance will suffer.
